### ISSUE TITLE
Cannot set property 'maxAge' of undefined

### ISSUE THREAD

**Issue by ButuzGOL** [koajs/session#35]
After update to version "koa-session": "^3.1.0" ("koa": "~0.18.1")
Getting this error

```
TypeError: Cannot set property 'maxAge' of undefined
      at new Session (/Users/butuzgol/Playground/game/node_modules/koa-session/index.js:152:59)
      at Object.session (/Users/butuzgol/Playground/game/node_modules/koa-session/index.js:64:16)
      at Object.properties.(anonymous function).get (/Users/butuzgol/Playground/game/node_modules/koa-passport/lib/framework/request.js:99:21)
      at Function.initialize (/Users/butuzgol/Playground/game/node_modules/koa-passport/node_modules/passport/lib/middleware/initialize.js:50:12)
      at /Users/butuzgol/Playground/game/node_modules/co/index.js:135:8
      at Object.thunkToPromise (/Users/butuzgol/Playground/game/node_modules/co/index.js:134:10)
      at Object.toPromise (/Users/butuzgol/Playground/game/node_modules/co/index.js:118:55)
      at next (/Users/butuzgol/Playground/game/node_modules/co/index.js:98:29)
      at onFulfilled (/Users/butuzgol/Playground/game/node_modules/co/index.js:68:7)
      at /Users/butuzgol/Playground/game/node_modules/co/index.js:53:5
```

On  "koa-session": "^2.0.0" works fine

**Comment by hiroqn** [comment]
koa-session v3 setup is like below

```
app.use(session(app))
```

v2 is
```
app.use(session())
```

**Comment by Ziink** [comment]
The problem shows up if there's already a cookie.

The code
`if ('_maxAge' == k) this._ctx.sessionOptions.maxAge = obj._maxAge1`
is what causes the problem because sessionOptions is never initialized for _ctx.

**Comment by Ziink** [comment]
Bug got fixed in 3.1.1 or 3.2.0
